pynecone/utils.py

```diff
@@ -799,41 +794,79 @@ def get_path_args(path: str) -> Dict[str, str]:
     Args:
         path: The path to get the arguments for.
 
-    Raises:
-        ValueError: explains what is wrong with the path.
-
     Returns:
         The path arguments.
     """
-    # Import here to avoid circular imports.
-    from pynecone.var import BaseVar
+
+    def add_path_arg(match, type_):
+        """Add arg from regex search result.
+
+        Args:
+            match: result of a regex search
+            type_: the assigned type for this arg
+
+        Raises:
+            ValueError: explains what is wrong with the path.
+        """
+        arg_name = match.groups()[0]
+        if arg_name in args:
+            raise ValueError(
+                f"arg name [{arg_name}] is used more than once in this URL"
+            )
+        args[arg_name] = type_
 
     # Regex to check for path args.
-    check = re.compile(r"^\[(.+)\]$")
-    check_catchall = re.compile(r"^\[\.\.\.(.+)\]$")
+    check = re.compile(constants.Regex.ARG)
```

Comment:
Maybe we can just compile these within the `constants` rather than doing it every time in this function. So the Enum contains the regexes instead of the str value. Not sure if that would complicate things though

Reply:
Yes it will make regexs faster and make startup time a little bit longer, but I have tested it and found compile regex for ten thousand times only cost less than 0.01s, so it is worth to do this.